util/validateparam.go

```diff
@@ -0,0 +1,27 @@
+package util
+
+import (
+	"fmt"
+	"log"
+
+	"strings"
+)
+
+func ValidateStringParam(paramName string, paramValue string, validValues []string) {
+	valueFound := false
+
+	for i := 0; i < len(validValues); i++ {
+		valueFound = (paramValue == validValues[i])
+		if valueFound { break }
+	}
+
+	if !valueFound {
+		err := fmt.Sprintf(
+			"Invalid value '%s' for param %s. Must be one of [%s].",
+			paramValue,
+			paramName,
+			strings.Join(validValues, ", "),
```

Comment:
You _might_ be able to give the `sprintf` the `validValues` array itself if you use `%v` as the formatting arg.

Reply:
Oh good point, I tried this and it prints the values without commas, which is fine as I think having a flag value with a space in it would probably be a code smell...